src/js/components/products/list.tsx

```diff
@@ -0,0 +1,197 @@
+
+import { EmptyIllustration } from '@/components/404';
+import { LoginButton } from '@/components/login';
+import ProductListItem from '@/components/products/listItem';
+import { LabelWithSpinner } from '@/components/utils';
+import { AppState } from '@/store';
+import { fetchMoreProducts, syncRepos } from '@/store/products/actions';
+import { Product } from '@/store/products/reducer';
+import { selectNextUrl, selectProducts } from '@/store/products/selectors';
+
+interface Props {
+  y: number;
+  products: Product[];
+  next: string | null;
+  doFetchMoreProducts({ url }: { url: string }): Promise<any>;
+  doSyncRepos(): Promise<any>;
+}
+
+const ProductList = withScroll(
+  ({ y, products, next, doFetchMoreProducts, doSyncRepos }: Props) => {
+    const [fetchingProducts, setFetchingProducts] = useState(false);
+    const [syncingRepos, setSyncingRepos] = useState(false);
+
+    const syncReposClicked = () => {
+      setSyncingRepos(true);
+      doSyncRepos().finally(() => {
+        setSyncingRepos(false);
+      });
+    };
+
+    useEffect(() => {
+      if (fetchingProducts || !next) {
+        return;
+      }
+
+      const maybeFetchMoreProducts = () => {
+        /* istanbul ignore else */
+        if (next && !fetchingProducts) {
+          setFetchingProducts(true);
+          doFetchMoreProducts({ url: next }).finally(() => {
+            setFetchingProducts(false);
+          });
+        }
+      };
+
+      /* istanbul ignore next */
+      const scrollHeight =
+        (document.documentElement && document.documentElement.scrollHeight) ||
+        (document.body && document.body.scrollHeight) ||
+        Infinity;
+      const clientHeight =
+        (document.documentElement && document.documentElement.clientHeight) ||
+        window.innerHeight;
+      // Fetch more products if within 100px of bottom of page...
+      const scrolledToBottom =
+        scrollHeight - Math.ceil(y + clientHeight) <= 100;
+
+      /* istanbul ignore else */
+      if (scrolledToBottom) {
+        maybeFetchMoreProducts();
+      }
+    }, [y, next, fetchingProducts, doFetchMoreProducts]);
+
+    let contents;
+    switch (products.length) {
+      case 0: {
+        // No products; show empty message
+        const msg = i18n.t(
+          'We couldn’t find any products you have access to on GitHub. Are you logged into the correct account?',
+        );
+        contents = (
+          <>
+            <EmptyIllustration message={msg} />{' '}
+            <div className="slds-align_absolute-center">
+              <LoginButton
+                id="auth-error-login"
+                label={i18n.t('Log In With a Different Account')}
+              />
+            </div>
+          </>
+        );
+        break;
+      }
+      default: {
+        contents = (
+          <div className="slds-grid slds-wrap slds-grid_pull-padded-small">
+            {products.map(product => (
+              <ProductListItem product={product} key={product.id} />
```

Comment:
I'm not clear  which github repos are being shown for what reasons. Maybe this is sample data, and I shouldn't worry about it?

Reply:
Products are shown if 1) they have been manually added to MetaShare by an admin, and 2) you are logged in and have contributor access on those repos.

This is sample data, with a number of SFDO and OddBird repos added.